### DISCUSSION TITLE
PyBullet: get dynamics information

### DISCUSSION THREAD

**Discussion by bdelhaisse** [bulletphysics/bullet3#3630] [General]
Hi Erwin,

Is it possible to get dynamics information when using PyBullet? Specifically, I would like to get:
- the joint accelerations when calling `getJointState(s)`.
- the link acceleration in the world frame when calling `getLinkState`.
- the moment around a contact point when calling `getContactPoints`.

If you don't have the time to implement it, could you point me how I can implement these features? 

Thanks!

**Comment by erwincoumans** [comment]
We don't explicitly keep track of link/joint accelerations and torque around the contacts.
For accelerations, you could manually save the previous link/joint velocity and compute the accelerations using finite differencing.
We also don't store the applied torques at the contact points, we only expose the forces (normal, and 2 lateral friction directional forces). It could be added to pybullet.getContactPoints.

Explaining how is just as much work as doing the actual implementation.
